ethereumj-core/src/main/java/org/ethereum/db/IndexedBlockStore.java

```diff
@@ -290,6 +291,30 @@ public synchronized long getMaxNumber(){
         return getListBlocksEndWithInner(hash, qty);
     }
 
+    @Override
+    public List<Block> listBlocks(byte[] fromHash, byte[] toHash) {
+        Block from = blocks.get(fromHash);
+        Block to = blocks.get(toHash);
+
+        // sanity checks
+        if (from == null || to == null || to.getNumber() < from.getNumber())
+            return Collections.emptyList();
+
+        long qty = to.getNumber() - from.getNumber() + 1;
+
+        List<Block> ret = new ArrayList<>();
+        Block block = to;
+        for (long i = 0; i < qty && block != null && !from.isEqual(block); ++i) {
+            ret.add(block);
+            block = blocks.get(block.getParentHash());
+        }
+
+        if (ret.size() > 0 && !ret.get(ret.size() - 1).isEqual(from))
```

Comment:
why this could happens?

Reply:
The latter may happen when `from` block is not an ancestor of `to` block. The former... I just couldn't make a look up into a list without being sure that it's not empty.